flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetWindowAggregate.scala

```diff
@@ -294,18 +319,29 @@ class DataSetWindowAggregate(
           namedProperties)
 
         mappedInput.groupBy(groupingKeys: _*)
-        .sortGroup(rowTimeFieldPos, Order.ASCENDING)
-        .reduceGroup(groupReduceFunction)
-        .returns(rowTypeInfo)
-        .name(aggregateOperatorName)
-        .asInstanceOf[DataSet[Any]]
+          .sortGroup(rowTimeFieldPos, Order.ASCENDING)
+          .reduceGroup(groupReduceFunction)
+          .returns(rowTypeInfo)
+          .name(aggregateOperatorName)
+          .asInstanceOf[DataSet[Any]]
+
+      } else {
+        // non-grouping window
+        val mapPartitionFunction = createDataSetWindowAggregationMapPartitionFunction(
+          window,
+          namedAggregates,
+          inputType,
+          rowRelDataType,
+          namedProperties,
+          isPreMapPartition = false)
+
+        mappedInput.sortPartition(rowTimeFieldPos, Order.ASCENDING).setParallelism(1)
+          .mapPartition(mapPartitionFunction).setParallelism(1)
```

Comment:
I think we can also use `.reduceGroup()` and a `GroupReduceFunction` here. Without `groupBy`, the `GroupReduceFunction` will be executed with parallelism 1.